### ISSUE TITLE
Simplify live reload SSE endpoint after Puma bug is fixed

### ISSUE THREAD

**Issue by ayushn21** [bridgetownrb/bridgetown#950]
While pairing with @ioquatix, we looked at the live reload code in Bridgetown to try and make it work with both Falcon and Puma.

Currently we're spawning a background thread to hold the connection, which is at best a hack, and at worst, wrong. We need to do this because Ctrl+C doesn't kill Puma without it. This turns out to be a bug in Puma itself. We've filed an issue here: https://github.com/puma/puma/issues/3569.

After this is fixed, we should be able to remove the background thread from the SSE endpoint without any issues.

**Comment by jaredcwhite** [comment]
We can close this now, yes?

**Comment by ayushn21** [comment]
Yep! I thought merging the PR would've automatically closed it :)
